### PR TITLE
Fix dlopen linking

### PR DESCRIPTION
**Pull request by TheTripleV**

### CODE REVIEW

**Comment by virtuald** [comment]
Hm? What error are you getting?

**Comment by TheTripleV** [comment]
Building robotpy-halsim-gui fails with rpy-build main on
```py
running build_dl
Traceback (most recent call last):
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\download.py", line 68, in download_and_extract_zip
    info = z.getinfo(src)
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\zipfile.py", line 1429, in getinfo
    raise KeyError(
KeyError: "There is no item named 'windows/x86-64/shared/halsim_gui.lib' in the archive"

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\vasis\Desktop\rpy20211\robotpy-halsim-gui\setup.py", line 5, in <module>
    setup()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\setup.py", line 196, in setup
    s.run()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\setup.py", line 190, in run
    _setup(**self.setup_kwargs)
  File "C:\Users\vasis\Desktop\rpy20211\venv\lib\site-packages\setuptools\__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\command\develop.py", line 8, in run
    develop.run(self)
  File "C:\Users\vasis\Desktop\rpy20211\venv\lib\site-packages\setuptools\command\develop.py", line 34, in run
    self.install_for_development()
  File "C:\Users\vasis\Desktop\rpy20211\venv\lib\site-packages\setuptools\command\develop.py", line 136, in install_for_development
    self.run_command('build_ext')
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\command\build_ext.py", line 110, in run
    self.run_command("build_gen")
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
    cmd_obj.run()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\command\build_gen.py", line 33, in run
    self.run_command("build_dl")
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\vasis\.pyenv\pyenv-win\versions\3.9.0\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\command\build_dl.py", line 47, in run
    wrapper.on_build_dl(self.build_cache, self.src_unpack_to)
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\wrapper.py", line 321, in on_build_dl
    libnames_full = self._clean_and_download(downloads, cache, srcdir)
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\wrapper.py", line 407, in _clean_and_download
    download_and_extract_zip(dl.url, to, cache)
  File "c:\users\vasis\desktop\rpy20211\robotpy-build\robotpy_build\download.py", line 78, in download_and_extract_zip
    raise ValueError(msg) from e
ValueError: error extracting windows/x86-64/shared/halsim_gui.lib from build\cache\halsim_gui-2021.1.2-windowsx86-64.zip
```

even though `pyproject.toml` has `dlopenlibs = ["halsim_gui"]`

The change was made in https://github.com/robotpy/robotpy-build/pull/88

**Comment by virtuald** [comment]
I feel like this must be in there for a reason... maybe it was the opencv libraries?

**Comment by TheTripleV** [comment]
Fixes:
1. Only autopopulate libs if both libs and dlopenlibs are unset or set to None
2. Actually extract dlopenlibs

File: robotpy_build/maven.py [hunk 1]
```diff
@@ -38,7 +38,10 @@ def convert_maven_to_downloads(
         raise ValueError("patches must be None if use_sources is False!")
     else:
         # libs
-        dl_lib["libs"] = mcfg.libs if mcfg.libs else [mcfg.artifact_id]
+
+        dl_lib["libs"] = mcfg.libs
+        if mcfg.libs == mcfg.dlopenlibs == None:
```
Comment (auscompgeek) [inline robotpy_build/maven.py:43]: Use `is` to compare to `None`.

**Pull request merged**